### PR TITLE
Version/update release creator

### PR DESCRIPTION
**Pull request by mickaelandrieu**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | We now manage the version of PrestaShop more like many popular projects, but we forgot to update the release manager.
| Type?         | bug fix
| Category?     | CO
| BC breaks?    | no
| Deprecations? | no
| How to test?  | Do `php tools/build/CreateRelease.php --version="1.7.2.4"` at the root of the project and check the `AppKernel` class is updated as expected. For others checks, I honestly don't know /c @eternoendless 

#### Important guidelines

* Make sure [your local branch is up to date](https://help.github.com/articles/syncing-a-fork/) before commiting your changes!
* Your code MUST respect [our Coding Standards](http://doc.prestashop.com/display/PS16/Coding+Standards) (for code written in PHP, JavaScript, HTML/CSS/Smarty/Twig, SQL)!

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/9289)

### CODE REVIEW

File: tools/build/CreateRelease.php [hunk 1]
```diff
@@ -27,6 +27,7 @@
 require_once 'Exception/BuildException.php';
 require_once 'Library/ReleaseCreator.php';
 require_once 'Library/ConsoleWriter.php';
+require_once 'Library/Version.php';
```
Comment (eternoendless) [inline tools/build/CreateRelease.php:30]: Looks like you forgot this file.

**Review by eternoendless — CHANGES_REQUESTED** [review]

File: tools/build/CreateRelease.php (hunk 1 above)
Comment (mickaelandrieu) [reply tools/build/CreateRelease.php:30]: True, maybe because this folder is ignored? I had to add -f it

File: tools/build/Library/Version.php [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+    /**
+     * Initialize version data
+     *
+     * @param string  $version  Version
+     */
+    public function __construct($version)
+    {
+        $this->version = $version;
+        $versions = explode('.', $version);
+        $this->majorVersionString = (string) $versions[0] . '.' . (string) $versions[1];
```
Comment (PierreRambaud) [inline tools/build/Library/Version.php]: explode already returns an array of string, so cast is useless.

File: tools/build/Library/Version.php [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+/**
+ * Class responsible of managing the right version of Shop
+ * for every internal/external services.
+ */
+class Version
+{
+    const STRING = 0;
```
Comment (PierreRambaud) [inline tools/build/Library/Version.php]: Don't understand why there are these const (STRING and INTEGER) :/ ?

Comment (mickaelandrieu) [reply tools/build/Library/Version.php]: Why do we have these constants in the first place? https://github.com/PrestaShop/PrestaShop/blob/develop/src/Core/Foundation/Version.php#L36 it's only a copy of the official class, but we cant rely on composer here :/

Comment (PierreRambaud) [reply tools/build/Library/Version.php]: And still don't understand why they are here :P

**Comment by mickaelandrieu** [comment]
Thanks for review @PierreRambaud, would you mind to test the release built or put the label `waiting for QA`?

What needs to be tested after the execution of the command:
* PrestaShop web installation process should works
* you should see the right version of PrestaShop in the login form of Back Office
* you should see the right version of PrestaShop in the debug toolbar on modern pages
* did I miss something? /c @eternoendless

**Comment by Quetzacoalt91** [comment]
@mickaelandrieu why didn't you create a symbolic link of the Version class to avoid maintainability issues later?

**Comment by mickaelandrieu** [comment]
@Quetzacoalt91 They are not strictly the same, look at the constructor.

**Comment by PierreRambaud** [comment]
And symbolic link not working on Windows (I think) :D

**Comment by Quetzacoalt91** [comment]
As there is only a different constructor, why didn't you only create a class which adapt the method params?

Something like 

```php
class Version extends PrestaShop\PrestaShop\Core\Foundation\Version
{
    /**
     * Initialize version data
     *
     * @param string  $version  Version
     */
    public static function createFromVersion($version)
    {
        return new self(
            $version,
            $versions[0] . '.' . $versions[1],
            (int) ($versions[0] . $versions[1]),
            $versions[2],
            $versions[3]
        );
    }
}
```

**Review by eternoendless — APPROVED** [review]

**Comment by eternoendless** [comment]
>  why didn't you only create a class which adapt the method params?

While it would be nice to reuse it, I'm not sure that it's a good idea to make the builder depend on the code it's building 🤪👈.

**Comment by Quetzacoalt91** [comment]
On the other side, we duplicate code without linking them together, not even with a comment.

I'm pretty sure one of them will be modified while the other will be forgotten. :)

**Comment by mickaelandrieu** [comment]
> PrestaShop\PrestaShop\Core\Foundation\Version

I cant' do that, @eternoendless forbid me to require anything outside of build folder :)

> I'm pretty sure one of them will be modified while the other will be forgotten. :)

As `build` folder needs to be independent of any other code, this is something we need to live with (sadly)

But as we will have soon nightly builds, we will be able to fix it quickly (I guess nightly will use tools like `ReleaseCreator`) and `Version` class from Core is unit tested.

**Comment by mickaelandrieu** [comment]
```php
class Version extends PrestaShop\PrestaShop\Core\Foundation\Version
{
    /**
     * Initialize version data
     *
     * @param string  $version  Version
     */
    public static function createFromVersion($version)
    {
        return new self(
            $version,
            $versions[0] . '.' . $versions[1],
            (int) ($versions[0] . $versions[1]),
            $versions[2],
            $versions[3]
        );
    }
}
```

looks like a good improvement of current class btw :)

**Comment by eternoendless** [comment]
Thank you @mickaelandrieu

**Pull request merged**